src/commonTest/kotlin/com/github/quillraven/fleks/SystemTest.kt

```diff
@@ -431,4 +438,97 @@ internal class SystemTest {
+
+        var check: Int = 0
+
+        val world = configureWorld {
+            families {
+                val family = family { all(NameComponent) }
+                onAdd(family) { entity ->
+                    assertEquals(0, check++) // First, we expect the global add hook to be called.
+                    println("$check. Global family onAdd() called with: entity = ${entity[NameComponent].name}")
+                }
+                onRemove(family) { entity ->
+                    assertEquals(5, check++) // The global hook gets called before the systems.
+                    println("$check. Global family onRemove() called with: entity = ${entity[NameComponent].name}")
+                }
+            }
+
+            systems {
+                add(object : IteratingSystem(family { all(NameComponent) }), IteratingSystem.FamilyOnAdd, IteratingSystem.FamilyOnRemove {
+                    val sysName = "System0"
+
+                    override fun onAddEntity(entity: Entity) {
+                        assertEquals(1, check++) // onAdd hooks called in forward system order. Thus, "System0" gets called first.
+                        println("$check. $sysName onAddEntity() called with: entity = ${entity[NameComponent].name}")
+                    }
+
+                    override fun onTickEntity(entity: Entity) {
+                        assertEquals(3, check++)
+                        println("$check. $sysName onTickEntity() called with: entity = ${entity[NameComponent].name}")
+                    }
+
+                    override fun onRemoveEntity(entity: Entity) {
+                        assertEquals(6, check++)
+                        println("$check. $sysName onRemoveEntity() called with: entity = ${entity[NameComponent].name}")
+                    }
+                })
+                add(object : IteratingSystem(family { all(NameComponent) }), IteratingSystem.FamilyOnAdd, IteratingSystem.FamilyOnRemove {
+                    val sysName = "System1"
+
+                    override fun onAddEntity(entity: Entity) {
+                        assertEquals(2, check++)
+                        println("$check. $sysName onAddEntity() called with: entity = ${entity[NameComponent].name}")
+                    }
+
+                    override fun onTickEntity(entity: Entity) {
+                        assertEquals(4, check++)
+                        println("$check. $sysName onTickEntity() called with: entity = ${entity[NameComponent].name}")
+                    }
+
+                    override fun onRemoveEntity(entity: Entity) {
+                        assertEquals(7, check++) // Lastly, the system with the highest order receives the hook call.
+                        println("$check. $sysName onRemoveEntity() called with: entity = ${entity[NameComponent].name}")
+                    }
+                })
+            }
+        }
+        val entity = world.entity { it += NameComponent("Entity0") }
```

Comment:
Make a new line before `val entity`. I forgot how it is called but the format that I am always following is

```kotlin
SETUP

EXECUTE

VERIFY
```

This is a known format and especially nice in shorter tests. You had all the setup before which will get shorter when println gets removed and systems are extracted.

Then you want to test the onAdd functionality (=new line).

I'd then either make a separate test for onRemove but keeping it in the same test is fine for me but I'd then reset your check counter and reformat to the following:

```kotlin
fun iterationSystemFamilyHookFlow() {
  // ... setup

  // verify onAdd
  val entity = world.entity { it += NameComponent("Entity0") }

  // verify onRemove
  check = 0
  world -= entity
}
```